src/osp/Active/SysVehicle.cpp

```diff
@@ -366,20 +365,11 @@ std::pair<ActiveEnt, std::vector<SysVehicle::MachineDef>> SysVehicle::part_insta
                 textureResources.push_back(texRes);
             }
 
-            // by now, the mesh and texture should both exist
-            using adera::shader::Phong;
-            Phong::ACompPhongInstance shader;
-            shader.m_shaderProgram = glResources.get<Phong>("phong_shader");
-            shader.m_textures = std::move(textureResources);
-            shader.m_lightPosition = Vector3{10.0f, 15.0f, 5.0f};
-            shader.m_ambientColor = 0x111111_rgbf;
-            shader.m_specularColor = 0x330000_rgbf;
-            rScene.reg_emplace<Phong::ACompPhongInstance>(currentEnt, std::move(shader));
-
-            CompDrawableDebug& bBocks = rScene.reg_emplace<CompDrawableDebug>(
-                        currentEnt, meshRes, &Phong::draw_entity, 0x0202EE_rgbf);
-
-            //new DrawablePhongColored(*obj, *m_shader, *mesh, 0xff0000_rgbf, m_drawables);
+            // by now, the mesh and texture should both exist so we emplace them
+            // as components to be consumed by the Phong shader
+            rScene.reg_emplace<ACompMesh>(currentEnt, meshRes);
+            rScene.reg_emplace<ACompDiffuseTex>(currentEnt, std::move(textureResources[0]));
```

Comment:
you move only the first texture? Why bother creating a vector if you are only ever using the first item in it?

Reply:
Temporary ugliness. Eventually we'll be loading several textures, and the code above this is sort of sitting around waiting to be hammered into a proper PBR resource loader. It was like that before this PR.

Reply:
understood